homeassistant/setup.py

```diff
@@ -733,6 +724,24 @@ def _setup_times(
     return hass.data[DATA_SETUP_TIME]  # type: ignore[no-any-return]
 
 
+def _setup_started(
+    hass: core.HomeAssistant,
+) -> dict[tuple[str, str | None], float]:
+    """Return the setup started dict."""
+    if DATA_SETUP_STARTED not in hass.data:
+        hass.data[DATA_SETUP_STARTED] = {}
+    return hass.data[DATA_SETUP_STARTED]  # type: ignore[no-any-return]
+
+
+def _setup_running(
+    hass: core.HomeAssistant,
+) -> set[tuple[str, str | None, SetupPhases]]:
+    """Return the setup running set."""
+    if DATA_SETUP_RUNNING not in hass.data:
+        hass.data[DATA_SETUP_RUNNING] = set()
+    return hass.data[DATA_SETUP_RUNNING]  # type: ignore[no-any-return]
```

Comment:
```suggestion
    return hass.data.setdefault(DATA_SETUP_RUNNING, set())  # type: ignore[no-any-return]
```
Wouldn't set default be a good use for setup_started and setup_running?

Reply:
setdefault makes a new `set()` every time and than throws it away